Sources/DesignSystem/Views/BraveTextEditor.swift

```diff
@@ -0,0 +1,38 @@
+// Copyright 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import SwiftUI
+
+/// A `TextEditor` made to look in the same way as `BraveTextFieldStyle` and adds a similar looking "prompt" (placeholder) to the view.
+public struct BraveTextEditor: View {
+  @Binding private var text: String
+  private let prompt: String
+  
+  public init(text: Binding<String>, prompt: String) {
+    _text = text
+    self.prompt = prompt
+  }
+  
+  public var body: some View {
+    TextEditor(text: $text)
+      .modifier(BaseBraveTextFieldStyleModifier(
+        strokeColor: nil, lineWidthFactor: nil,
+        paddingVertical: 2, paddingHorzontal: 6
+      ))
+      .overlay(
+        Text(prompt)
+          .disabled(true)
+          .allowsHitTesting(false)
+          .font(.callout)
+          .padding(.vertical, 10)
+          .padding(.horizontal, 12)
+          .frame(maxWidth: .infinity, alignment: .leading)
+          .foregroundColor(Color(.placeholderText))
+          .opacity(text.isEmpty ? 1 : 0)
+          .accessibilityHidden(true),
```

Comment:
The prompt should not be hidden from VoiceOver unless text is non-empty

Reply:
Good call